pkg/constraints/constraint.go

```diff
@@ -0,0 +1,83 @@
+package constraints
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// OLMConstraintType is the schema "type" key for all constraints known to OLM
+// (except for legacy types).
+const OLMConstraintType = "olm.constraint"
+
+// Constraint holds parsed, potentially nested dependency constraints.
+type Constraint struct {
+	// Constraint message that surfaces in resolution
+	// This field is optional
+	Message string `json:"message"`
```

Comment:
Every other field has both yaml and json tags defined, should this field include a yaml tag? Applies everywhere.

Reply:
This is user configurable eh? Slightly concerned what could be written here, but it should be fine.

Reply:
Huh not sure where the yaml tag went, will re-add.